unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -7957,23 +7805,40 @@ public Vector3 GetSurfacePointBelowPosition(Vector3 position)
 
         // End code for calculating the volume of a mesh
 
-        public void RandomlyOpenCloseObjects(int randomSeed = 0, bool simplifyPhysics = false) {
-            System.Random rnd = new System.Random(randomSeed);
-            List<CanOpen_Object> toInteractWith = new List<CanOpen_Object>();
+        // @pOpen is the probability of opening an openable object.
+        // @randOpenness specifies if the openness for each opened object should be random, between 0% : 100%, or always 100%.
+        public void RandomlyOpenCloseObjects(
+                int? randomSeed = null,
+                bool simplifyPhysics = false,
+                float pOpen = 0.5f,
+                bool randOpenness = true
```

Comment:
Note that there are 2 potentially breaking changes with this beta action:
* `randomSeed` is not set to 0 by default. Hence, if it is not passed in, the objects that are opened are actually random in the same scene.
* `randOpenness` specifies if the newly opened openness values should be randomized in `[0:1]` or if every openness should be exactly 1.

Note that if there is a random seed and `randOpenness == True`, each chosen openness value will also be sampled from a random seed.